### PR TITLE
Components and Filter for ConfigurableEnum values

### PR DESCRIPTION
**Pull request by sleidig**
required for adaptions for new customers

### Architectural/Backend Changes
- [x] add a filter type `"configurable-enum"` (see `ConfigurableEnumFilterConfig`) to display filter options in lists based on an enum in the config
- [x] refactor entity-list to extract filter component template
- [x] add field types for form (`input === 'configurable-enum-select'"`) and list (`DisplayConfigurableEnumComponent`) to display ConfigurableEnumValues

> please merge without sqashing

### CODE REVIEW

File: src/app/child-dev-project/attendance/attendance-analysis/attendance-analysis.component.ts [hunk 1]
```diff
@@ -95,10 +95,11 @@ export class AttendanceAnalysisComponent implements OnInit, AfterViewInit {
 
   applyFilterSelections() {
     this.loading = 0;
-    let filteredData = this.childrenAll;
+    const filteredData = this.childrenAll;
 
+    // TODO reactive filters or rewrite
     this.filterSelections.forEach((f) => {
-      filteredData = filteredData.filter(f.getSelectedFilterFunction());
+      // filteredData = filteredData.filter(f.getSelectedFilterFunction());
```
Comment (sleidig) [inline src/app/child-dev-project/attendance/attendance-analysis/attendance-analysis.component.ts:102]: this component will be completely replace due to the event attendance changes. therefore no fixes here

File: src/app/core/entity-components/entity-list/entity-list.component.ts [hunk 2]
```diff
@@ -259,9 +275,14 @@ export class EntityListComponent<T extends Entity>
...
       case "prebuilt":
         return (filter.options = (config as PrebuiltFilterConfig<T>).options);
+      case "configurable-enum":
+        return (filter.options = this.createConfigurableEnumFilterOptions(
+          config as ConfigurableEnumFilterConfig<T>
+        ));
       default: {
         const options = [...new Set(this.entityList.map((c) => c[config.id]))];
         filter.initOptions(options, config.id);
+        return filter.options;
```
Comment (TheSlimvReal) [inline src/app/core/entity-components/entity-list/entity-list.component.ts:285]: I think I made a mistake concerning the return type here. The other options only return `null`

**Review by TheSlimvReal — APPROVED** [review]
Very nice change, cleaned up a lot of things. I just found a little type issue (that I implemented myself earlier). please have a look at the changes that I added.

File: src/app/core/entity-components/entity-list/list-filter/list-filter.component.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<div *ngIf="displayAsToggle">
```
Comment (TheSlimvReal) [inline src/app/core/entity-components/entity-list/list-filter/list-filter.component.html:1]: The styling is not very good yet

filters_screenshot

**Review by TheSlimvReal — CHANGES_REQUESTED** [review]
Just some styling issues

**Review by TheSlimvReal — APPROVED** [review]
Looks all fine. Good work!

**Pull request merged**